### PR TITLE
BUG: ArrayManager reindex with copy=True not copying

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/methods/test_reindex.py [hunk 1]
```diff
@@ -83,6 +83,16 @@ class TestDataFrameSelectReindex:
...
     # test_indexing
 
+    def test_reindex_copies(self):
+        # based on asv time_reindex_axis1
+        N = 10
+        df = DataFrame(np.random.randn(N * 10, N))
+        cols = np.arange(N)
+        np.random.shuffle(cols)
+
+        result = df.reindex(columns=cols, copy=True)
```
Comment (jreback) [inline pandas/tests/frame/methods/test_reindex.py:93]: can you check with both axis reindexing as well (result should be the same) e.g. only a single copy.

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_reindex.py:93]: do you mean calling `df.reindex(index=foo, copy=True)`?  that's covered by existing tests.  this test should be focused on the particular bug

Comment (jreback) [reply pandas/tests/frame/methods/test_reindex.py:93]: yeah actually meant calling them *both* (columns AND index)

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_reindex.py:93]: updated + greenish

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks!

**Comment by jreback** [comment]
thanks, can you add on a whatsnew note at some point.

**Pull request merged**